server/medication/router.ts

```diff
@@ -0,0 +1,117 @@
+import type { Request, Response } from 'express';
+import express from 'express';
+import MedicationCollection from './collection';
+import * as medicationValidator from './middleware';
+import * as accountValidator from '../account/middleware';
+import * as util from './util';
+
+const router = express.Router();
+
+/**
+ * Get all of the session account's medications
+ *
+ * @name GET /api/medications
+ *
+ * @return {MedicationResponse[]} - An array of medications created by the user in alphabetical order with respect to the name of the medication
+ * 
+ * @throws {401} - If the user is not logged in
+ */
+router.get(
+  '/',
+  accountValidator.isLoggedIn,
+  async (req: Request, res: Response) => {
+    const medicalContacts = await MedicationCollection.findAllByOwner(req.session.credentialId as string);
+    res.status(200).json(medicalContacts.map(util.constructMedicationResponse));
+  }
+);
+
+/**
+ * Create a new medication record.
+ *
+ * @name POST /api/medication
+ * 
+ * @param {string} name - The username of the user
```

Comment:
[nit] I believe all the param specifications need to be updated for this function and the following functions

Reply:
thanks I'll fix that now. Just FYI this is also the case in all of your routes (I copied from there and it looks like we both missed it until now)